cmd/docker/docker.go

```diff
@@ -105,3 +111,23 @@ func dockerPreRun(opts *cliflags.ClientOptions) {
 		utils.EnableDebug()
 	}
 }
+
+func hideExperimentalFeatures(cmd *cobra.Command) {
+	for _, ccmd := range cmd.Commands() {
+
+		// hide commands
+		name := strings.Split(ccmd.Use, " ")[0]
+		if name == "stack" || name == "deploy" || name == "checkpoint" || name == "plugin" {
+			ccmd.Hidden = true
+		}
+
+		// hide flags
+		ccmd.Flags().VisitAll(func(f *pflag.Flag) {
```

Comment:
Yes, we can do that in our fork of cobra (I suggested that for the other option as well https://github.com/docker/docker/pull/28008#issuecomment-258216543).

So it sounds like either option would benefit from this.